packages/odata-service-inquirer/src/prompts/connectionValidator.ts

```diff
@@ -259,7 +259,11 @@ export class ConnectionValidator {
                 // Full service URL
                 await this.createOdataServiceConnection(axiosConfig, url.pathname);
             }
-            this._validatedClient = url.searchParams.get(SAP_CLIENT_KEY) ?? undefined;
+            this._validatedClient =
+                url.searchParams.get(SAP_CLIENT_KEY) ??
+                this.getSapClient(this._serviceProvider?.cookies?.toString()) ??
```

Comment:
The only supported client setting is through the search params. Therefore this code should be unnecessary.